force-app/main/default/classes/DMLOperations.cls

```diff
@@ -198,10 +259,15 @@ public with sharing class DMLOperations {
      */
     public static void insertAndDeleteLeads(List<String> leadNames) {
         // Loop through each Lead name
-
+        List<Lead> leadsList = new List<Lead>();
+        for(String s : leadNames){
+            leadsList.add(new Lead(LastName = s, Company = s));
+        }
```

Comment:
Again, avoid those single-letter variable names. Here, use `name`. This tells us instantly what we're dealing with:
```suggestion
        for(String name : leadNames){
            leadsList.add(new Lead(LastName = name, Company = name));
        }
```